src/dbnode/storage/bootstrap/result/shard_ranges.go

```diff
@@ -118,15 +118,47 @@ func (r shardTimeRanges) Equal(other ShardTimeRanges) bool {
 
 // IsSuperset returns whether the current shard time ranges is a superset of the
 // other shard time ranges.
+//
+// The following are superset conditions.
+// - Always superset if other shard time ranges are empty. (Even empty vs empty).
+// - Shards must be a superset of shards in other.
+// - Time ranges must be a superset of time ranges in other.
+//
+// The following are superset failure conditions:
+// - Partially overlapping shards.
+// - Non overlapping shards.
+// - Partially overlapping time ranges.
+// - Non overlapping time ranges.
 func (r shardTimeRanges) IsSuperset(other ShardTimeRanges) bool {
+	// If other shard time ranges is empty then the curr shard time ranges
+	// is considered a superset.
+	if other.Len() == 0 {
+		return true
+	}
+
+	// Short-circuit if we have more shards than other.
```

Comment:
Do we still need the short circuit? Maybe keep this method as short as possible since already complicated and rely on the new .Get(...) logic to check this shard time ranges has all that the other has?